.github/workflows/codequality.yml

```diff
@@ -21,18 +21,9 @@ permissions:
   pull-requests: read
 
 jobs:
-  lint:
-    name: lint
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/setup-go@v5
-        with:
-          go-version: "1.21"
-          cache: false
-      - uses: actions/checkout@v4
-      - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3
-        with:
-          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
-          version: v1.55
-          args: --timeout=5m
+  code-linter:
+    name: "Run golangci-lint"
+    uses: kyma-project/eventing-tools/.github/workflows/lint-go-reusable.yml@main
+    with:
+      go-version: "1.21"
```

Comment:
In NATS-Manager we use Go version [`1.22.0`](https://github.com/kyma-project/nats-manager/blob/5e038645a90b1aed9ec4b96e8c06e87caa77a91c/go.mod#L3). Why would we set the linter to `1.21`?

Instead of doing [this](https://github.com/kyma-project/eventing-tools/blob/abb93fa301d4b1aa07b46e5b01ffb0463f06e1bb/.github/workflows/lint-go-reusable.yml#L26) we could to this: 

```yaml
  - uses: actions/setup-go@v5
    with:
      go-version-file: 'go.mod'
```

Because the used Go version should always be what we use in the mod file anyhow. Otherwise, whenever Dependabot bumps the Go version in the mod file we would be required to (not forget to) manually adopt the workflows Go version.

Reply:
I did not change the Go version because I had no information abut the version change.

Reply:
We use `go.mod` here: https://github.com/kyma-project/eventing-tools/blob/abb93fa301d4b1aa07b46e5b01ffb0463f06e1bb/.github/workflows/lint-reusable.yml#L17

Reply:
To use 'go.mod' we need to modify the entire centralized method in eventing-tool repo. I suggest to do it in a separated way and not in this PR.

Reply:
> I did not change the Go version because I had no information abut the version change.

That is exactly the problem with this. We will be required to do it in two places, and it is easy to miss changes in one place that would be needed to transfer over in another place.

> To use 'go.mod' we need to modify the entire centralized method in eventing-tool repo. I suggest to do it in a separated way and not in this PR.

Sounds reasonable.